pkg/proctree/process.go

```diff
@@ -0,0 +1,383 @@
+}
+
+// connectParent add given process as the parent process of the current one
+func (p *processNode) connectParent(parent *processNode) {
+	p.parentProcess = parent
+}
+
+// connectChild add given process as the child process of the current one
+func (p *processNode) connectChild(child *processNode) {
+	p.childProcesses[child.getId()] = child
+}
+
+// This doesn't have to be protected by mutex, as the process Id shouldn't change after creation
+func (p *processNode) getId() int {
+	return p.id
+}
+
+// getNsId return the PID of the process in its PID namespace
+func (p *processNode) getNsId() int {
+	return p.nsId
+}
+
+// setNsId set the PID of the process in its namespace to given one
+func (p *processNode) setNsId(nsId int) {
+	p.nsId = nsId
+}
+
+// getContainerId return the ID of the container in which the process resides
+func (p *processNode) getContainerId() string {
+	return p.containerId
+}
+
+// setContainerId set the ID of the container in which the process resides
+func (p *processNode) setContainerId(containerId string) {
+	p.containerId = containerId
+}
+
+// getUserId return the ID of the user owning the process
+func (p *processNode) getUserId() int {
+	return p.userId
+}
+
+// setUserId set the ID of the user owning the process
+func (p *processNode) setUserId(userId int) {
+	p.userId = userId
+}
+
+// getThread return the thread with given TID if is a registered thread of the process
+func (p *processNode) getThread(tid int) (*threadNode, bool) {
+	thread, ok := p.threads[tid]
+	return thread, ok
+}
+
+// connectToThread add reference to given thread as a thread of the current process
+func (p *processNode) connectToThread(thread *threadNode) {
+	p.threads[thread.getId()] = thread
+}
+
+// disconnectThread remove the reference to given thread from the current process
+func (p *processNode) disconnectThread(thread *threadNode) {
+	delete(p.threads, thread.getId())
+}
+
+// getThreads return all the registered threads of current process
+func (p *processNode) getThreads() []*threadNode {
+	return maps.Values(p.threads)
+}
+
+// getThreadsIds return the TIDs of all registered thread of current process
+func (p *processNode) getThreadsIds() []int {
+	return maps.Keys(p.threads)
+}
+
+// getChild return the child process with given PID if registered as a child of the current process
+func (p *processNode) getChild(pid int) (*processNode, bool) {
+	child, ok := p.childProcesses[pid]
+	return child, ok
+}
+
+// getChildren return all registered children processes of current process
+func (p *processNode) getChildren() []*processNode {
+	return maps.Values(p.childProcesses)
+}
+
+// amountOfChildren return the amount of processes registered as children of current process
+func (p *processNode) amountOfChildren() int {
+	return len(p.childProcesses)
+}
+
+// hasChildren return if the current process has children registered to it
+func (p *processNode) hasChildren() bool {
+	return p.amountOfChildren() != 0
+}
```

Comment:
In here.. the len(p.childProcesses) read the metadata for the map (to get its length). At the same time, you have `maps.Values` walking the map to construct a slice (and its not thread-safe). Those 2 aren't thread-safety among them (lets say you can `amountOfChildre()` in 2 different threads). Now the question: "Oh but I won't be calling methods from 2 different threads because its always coming from the control plane". Then there is no need for thread-safety at all. I strongly think all our structs/classes should be thread-safe for all its operations to all its fields/transactions... arguable and I know others disagree.